cmd/autoconfigbrancher/main.go

```diff
@@ -89,44 +87,35 @@ func validateOptions(o options) error {
 	return o.GitHubOptions.Validate(o.dryRun)
 }
 
-func hasChanges() (bool, error) {
-	cmd := "git"
-	args := []string{"status", "--porcelain"}
-	logrus.WithField("cmd", cmd).WithField("args", args).Info("running command ...")
-	combinedOutput, err := exec.Command(cmd, args...).CombinedOutput()
-	if err != nil {
-		logrus.WithField("cmd", cmd).Debugf("output is '%s'", string(combinedOutput))
-		return false, err
+func runAndCommitIfNeeded(stdout, stderr io.Writer, author, cmd string, args []string) (bool, error) {
+	fullCommand := fmt.Sprintf("%s %s", filepath.Base(cmd), strings.Join(args, " "))
+
+	logrus.Infof("Running: %s", fullCommand)
+	if err := bumper.Call(stdout, stderr, cmd, args...); err != nil {
+		return false, fmt.Errorf("failed to run %s: %v", fullCommand, err)
 	}
-	return len(strings.TrimSuffix(string(combinedOutput), "\n")) > 0, nil
-}
 
-func commitIfNeeded(msg, author string) {
-	changed, err := hasChanges()
+	changed, err := bumper.HasChanges()
 	if err != nil {
-		logrus.WithError(err).Fatal("error occurred when checking changes")
+		return false, fmt.Errorf("error occurred when checking changes: %v", err)
 	}
-	if changed {
-		count++
-		addAndCommit(msg, author)
+
+	if !changed {
+		logrus.WithField("command", fullCommand).Info("No changes to commit")
+		return false, nil
 	}
-}
 
-func addAndCommit(msg, author string) {
-	cmd := "git"
-	args := []string{"add", "."}
-	run(cmd, args...)
-	cmd = "git"
-	args = []string{"commit", "-m", msg, "--author", author}
-	run(cmd, args...)
-}
+	gitCmd := "git"
+	if err := bumper.Call(stdout, stderr, gitCmd, []string{"add", "."}...); err != nil {
+		return false, fmt.Errorf("failed to 'git add .': %v", err)
+	}
 
-func run(cmd string, args ...string) {
-	logrus.WithField("cmd", cmd).WithField("args", args).Info("running command ...")
-	if combinedOutput, err := exec.Command(cmd, args...).CombinedOutput(); err != nil {
-		logrus.WithField("cmd", cmd).Debugf("output is '%s'", string(combinedOutput))
-		logrus.Fatalf("Failed to run command:'%s'", err)
+	commitArgs := []string{"commit", "-m", fullCommand, "--author", author}
+	if err := bumper.Call(stdout, stderr, gitCmd, commitArgs...); err != nil {
```

Comment:
While we are at it, can we make the whole thing one commit rather than seven?

Reply:
I think it's better to have each command in its commit.

Reply:
In the past it was useful to see the changes in separate commits, so I'd like to keep it that way, too.